File: problem1/problem1.cpp
#include <iostream>

// This function finds the sum of all multiples of the integer "find_multiples_of".
// All found multiples should be below the integer "upper_limit".
int sum_of_multiples(int find_multiples_of, int upper_limit)
{
    int sum = 0;
    int multiplier = 1;
    while(multiplier * find_multiples_of < upper_limit)
    {
        sum += multiplier * find_multiples_of;
        ++multiplier;
    }
    return sum;
}

int sum_of_common_multiples(int a, int b, int upper_limit)
{
    int larger_integer  = (a > b) ? (a) : (b);
    int smaller_integer = (larger_integer == a) ? (b) : (a);
    int multiple_of_larger_integer = 0;
    int sum = 0;
    int multiplier = 1;

    while(multiplier * larger_integer < upper_limit)
    {
        multiple_of_larger_integer = multiplier * larger_integer;
        if (multiple_of_larger_integer % smaller_integer == 0)
        {
            sum += multiple_of_larger_integer;
        }

        ++multiplier;
    }

    return sum;
}

int main()
{
    int upper_limit = 1000;
    std::cout << "The sum of all multiples of 3 or 5 below " << upper_limit << " is " 
              << (sum_of_multiples(3, upper_limit) + sum_of_multiples(5, upper_limit) - sum_of_common_multiples(3, 5, upper_limit))
              << std::endl;
    return 0;
}

File: problem1/problem1.py
# This function will find the sum of the 
# multiples of given 'integer', such that
# all multiples are below upper_limit
def sum_of_multiples(integer, upper_limit):
    multiplier = 1
    sum = 0
    multiple = integer * multiplier
    while multiple < upper_limit:
        sum += multiple
        multiplier += 1
        multiple = integer * multiplier
    return sum

def sum_of_common_multiples(integer1, integer2, upper_limit):
    multiplier = 1
    sum = 0
    larger_integer  = integer1 if integer1 > integer2 else integer2
    smaller_integer = integer1 if integer1 < integer2 else integer2
    multiple = larger_integer * multiplier
    while multiple < upper_limit:
        if multiple % smaller_integer == 0:
            sum += multiple
        multiplier += 1
        multiple = larger_integer * multiplier
    return sum

given_upper_limit = 1000
print(f"Sum of the multiples of 3 or 5 below 1000 is {sum_of_multiples(3, given_upper_limit) + sum_of_multiples(5, given_upper_limit) - sum_of_common_multiples(3, 5, given_upper_limit)}")